File: Java/src/main/java/com/ppdai/open/core/PropertyObject.java
package com.ppdai.open.core;

import java.text.ParseException;

/**
 * 属性对象
 * Created by xuzhishen on 2016/5/10.
 */
public class PropertyObject {

    private String name;
    private String lowerName;

    private Object value;
    private ValueTypeEnum valueType;

    /**
     * 属性对象
     * @param name  属性名称
     * @param value 属性值
     * @param valueType 属性类型
     */
    public PropertyObject(String name, Object value, ValueTypeEnum valueType)
    {
        this.name = name;
        this.lowerName = name.toLowerCase();
        this.value = value;
        this.valueType = valueType;
    }

    /**
     * 获取属性名称
     * @return
     */
    public String getName() {
        return name;
    }

    /**
     * 获取小写的属性名称
     * @return
     */
    public String getLowerName() {
        return lowerName;
    }

    /**
     * 获取属性值
     * @return
     */
    public Object getValue() {
        return value;
    }

    /**
     * 是否参与签名
     * @return
     */
    public boolean isSign(){
        return value != null && valueType != ValueTypeEnum.Other;
    }

    @Override
    public String toString() {
        try {
            return PropertyFormater.ObjectFormat(lowerName, value, valueType);
        } catch (ParseException e) {
            e.printStackTrace();
        }
        return  null;
    }
}

File: C#/OpenApi.Utility/PropertyObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OpenApi.Utility
{
    /// <summary>
    /// 键值对象
    /// </summary>
    public class PropertyObject
    {
        private string name;
        private string lowerName;

        private object value;
        private ValueTypeEnum valueType;

        /// <summary>
        /// 构造函数
        /// </summary>
        /// <param name="name">属性名称</param>
        /// <param name="value">属性值</param>
        /// <param name="valueType">属性类型</param>
        public PropertyObject(string name, object value, ValueTypeEnum valueType)
        {
            this.name = name;
            this.lowerName = name.ToLower();
            this.value = value;
            this.valueType = valueType;
        }

        /// <summary>
        /// 是否参与签名验证
        /// </summary>
        public bool IsSign {
            get { return value != null && valueType != ValueTypeEnum.Other; }
        }

        /// <summary>
        /// 获取小写名称
        /// </summary>
        public string LowerName {
            get { return lowerName; }
        }

        /// <summary>
        /// 获取名称
        /// </summary>
        public string Name {
            get { return name; }
        }

        /// <summary>
        /// 获取值
        /// </summary>
        public object Value {
            get { return value; }
        }

        /// <summary>
        /// 生成格式化字符串
        /// </summary>
        /// <returns></returns>
        public override string ToString()
        {
            return PropertyFormater.ObjectFormat(lowerName, value, valueType);
        }
    }
}
